src/layers/generic.cpp

```diff
@@ -283,31 +282,7 @@ namespace marian {
       };
 
       if (shortlist_ && !cachedShortWt_) { // shortlisted versions of parameters are cached within one batch, then clear()ed
```

Comment:
How difficult is to have the shortlist_ be passed to the Affine routine, then we can do it on the fly without extra cached nodes?

Reply:
Something to think about, but not in this PR maybe? This whole shortlist business is quite horrible already, having to pass even further down makes it worse IMO. This would benefit from a cleaner, better thought out approach.